### PR TITLE
Fix empty keyed encoding

### LINKED ISSUE #122: Keyed types are not encoded as JSONB when all properties are nil
When encoding a structure with all properties set to `nil`, `PostgreSQLDataEncoder` returns `.null` instead of an empty JSONB object.

The following test fails:

```swift
func testRowCodableEmptyKeyed() throws {
    let components = DateComponents()
    let row = try PostgreSQLDataEncoder().encode(components)
    XCTAssert(row.type == .jsonb)
}
```

Whereas this one passes:

```swift
func testRowCodableEmptyKeyed() throws {
    let components = DateComponents(hour: 10)
    let row = try PostgreSQLDataEncoder().encode(components)
    XCTAssert(row.type == .jsonb)
}
```

Because `nil` properties are skipped, the date components do not try to encode any key. Therefore the encoder fails to detect that it should use `JSONEncoder`.

### PR DESCRIPTION
**Pull request by baarde**
This fixes #122.

### CODE REVIEW

File: Tests/PostgreSQLTests/PostgreSQLConnectionTests.swift [hunk 1]
```diff
@@ -332,6 +332,12 @@ class PostgreSQLConnectionTests: XCTestCase {
         print(row)
     }
 
+    func testRowCodableEmptyKeyed() throws {
```
Comment (tanner0101) [inline Tests/PostgreSQLTests/PostgreSQLConnectionTests.swift]: Can you add this to the `allTests` array at the bottom of the file?

Comment (baarde) [reply Tests/PostgreSQLTests/PostgreSQLConnectionTests.swift]: Oops. I always forget `allTests`. Fixed.

**Comment by tanner0101** [comment]
Thanks :)

**Comment by penny-coin** [comment]
Hey @baarde, you just merged a pull request, have a coin! 

You now have 8 coins.

**Pull request merged**